File: java/TestCase.java
public class TestCase {
    public Tree[] t;
    public TestCase() {
        t = new Tree[] {
            new BranchS(new BranchM(new Leaf(2), new Leaf(4)), new Leaf(3)),
            new BranchM(new BranchS(new Leaf(2), new Leaf(4)), new Leaf(3)),
            new BranchS(new BranchS(new Leaf(1), new Leaf(2)), new Leaf(3)),
            new BranchS(new Leaf(1), new BranchS(new Leaf(2), new Leaf(3))),
            new BranchM(new BranchM(new Leaf(1), new Leaf(2)), new Leaf(3)),
            new BranchM(new Leaf(1), new BranchM(new Leaf(2), new Leaf(3))),
            new BranchM(new BranchS(new Leaf(1), new Leaf(2)), new BranchS(new Leaf(3), new Leaf(4))),
            new BranchM(new Leaf(6), new Leaf(7)),
            new BranchS(new BranchM(new Leaf(1), new Leaf(2)), new BranchM(new Leaf(3), new Leaf(4))),
            new BranchM(new BranchS(new Leaf(1), new BranchM(new Leaf(2), new Leaf(3))), new BranchM(new BranchM(new Leaf(4), new Leaf(5)), new BranchS(new Leaf(6), new Leaf(7)))),
            // 課題 1-1-3
            new BranchS(new Leaf(2), new BranchX(new Leaf(2), new BranchM(new Leaf(2), new Leaf(3)))),
            // 課題 1-1-4
            new BranchX(new Leaf(3), new BranchX(new Leaf(3), new Leaf(3))),
            new BranchX(new BranchX(new Leaf(3), new Leaf(3)), new Leaf(3)),
            // 課題 1-1-7 用(t が偶数個の要素を持たないといけないみたい)
            new BranchS(new Leaf(1), new Leaf(2))
        };
    }
}

File: ocaml/testcase.ml
open Tree

let test_case =
  [ BranchS (BranchM (Leaf 2, Leaf 4), Leaf 3);
    BranchM (BranchS (Leaf 2, Leaf 4), Leaf 3);
    BranchS (BranchS (Leaf 1, Leaf 2), Leaf 3);
    BranchS (Leaf 1, BranchS (Leaf 2, Leaf 3));
    BranchM (BranchM (Leaf 1, Leaf 2), Leaf 3);
    BranchM (Leaf 1, BranchM (Leaf 2, Leaf 3));
    BranchM (BranchS (Leaf 1, Leaf 2), BranchS (Leaf 3, Leaf 4));
    BranchM (Leaf 6, Leaf 7);
    BranchS (BranchM (Leaf 1, Leaf 2), BranchM (Leaf 3, Leaf 4));
    BranchM (BranchS (Leaf 1, BranchM (Leaf 2, Leaf 3)), BranchM (BranchM (Leaf 4, Leaf 5), BranchS (Leaf 6, Leaf 7)));
    (* 課題 1-1-3 *)
    BranchS (Leaf 2, BranchX (Leaf 2, BranchM (Leaf 2, Leaf 3)));
    (* 課題 1-1-4 *)
    BranchX (Leaf 3, BranchX (Leaf 3, Leaf 3));
    BranchX (BranchX (Leaf 3, Leaf 3), Leaf 3);
    (* 課題 1-1-7 用(t が偶数個の要素を持たないといけないみたい) *)
    BranchS (Leaf 1, Leaf 2)
  ]

let rec print_list f l i =
  match l with
    [] -> ()
  | hd::tl ->
     print_int i;
     print_string ": ";
     print_endline (f hd);
     print_list f tl (i+1)
let print_test_case f =
  print_list f test_case 0
